plugins/Diagnostics/Diagnostic/PhpSettingsCheck.php

```diff
@@ -55,22 +51,22 @@ public function execute()
     }
 
     /**
-     * @return string[]
+     * @return RequiredPhpSetting[]
      */
     private function getRequiredSettings()
     {
-        $requiredSettings = array(
-            // setting = required value
-            // Note: value must be an integer only
-            'session.auto_start=0',
-        );
+        $requiredSettings[] = new RequiredPhpSetting('session.auto_start', 0);
+        
+        $maxExecutionTime = new RequiredPhpSetting('max_execution_time', 0);
+        $maxExecutionTime->addRequiredValue(30, '>=');
+        $requiredSettings[] = $maxExecutionTime;
```

Comment:
Tested locally and I noticed that since required settings are added as an ERROR status, this will block installation if max_execution_time is not > 30 (eg, if it's set to 10 on some shared hosting provider, you wouldn't be able to install matomo).

@mattab do you think this is a good thing?

Reply:
@diosmosis +1 to make it a warning.